### PR TITLE
Propagate deletations and restoriations when using ModelTrait

### PR DESCRIPTION
**Pull request by pellaras**
When using ModelTrait on a model, deletions and restorations will now be reflected on DynamoDb, keeping it in sync with the local db.

### CODE REVIEW

File: src/ModelObserver.php [hunk 1]
```diff
@@ -45,4 +45,38 @@ public function saved($model)
             Log::info($e);
```
Comment (baopham) [inline src/ModelObserver.php]: Could you help change this to `Log::error`? `Log::info` isn't a good choice.

File: src/ModelObserver.php [hunk 2]
```diff
@@ -32,7 +32,7 @@ public function __construct(DynamoDbClientInterface $dynamoDb)
         $this->attributeFilter = $dynamoDb->getAttributeFilter();
     }
 
-    public function saved($model)
+    protected function _save($model)
```
Comment (baopham) [inline src/ModelObserver.php]: I would prefer to stay away from underscore prefix in method name. Could you:

1. Change this to `private`
2. Rename it to `saveToDynamoDb`

Same for the `_delete` method.

**Comment by pellaras** [comment]
Of course! It's the fastest way to get up and running! :D

Made the changes ;)

**Comment by baopham** [comment]
Thank you!

**Comment by baopham** [comment]
Released in v4.11.0

**Pull request merged**